### PR TITLE
Cli Repl: Automatically resolve promises and add last result variable

### PR DESCRIPTION
**Pull request by sam-finch-tezos**
# Context
The CLI repl does not support the `await` keyword at the lop level.
This PR turns the bug into a feature by adding the following behaviour
* If the repl command returns a promise the. repl will block until the promise is resolved and returns the fulfilment value of the promise
* The result of the last run command, or the fulfilment value in the case of a promise, is stored in a variable `_` for easy reuse.

**Related Tasks**: [REPL: Resolve promises automatically](https://app.asana.com/0/1205170694555856/1205666947698889/f)

# Manually testing the PR
```javascript
cargo run -- repl
> Contract.create("export default () => { console.log(`Hello from ${Ledger.selfAddress} 👋`); return new Response() }");
> new Request(`tezos://${_}`);
> Contract.call(_);
```

### CODE REVIEW

**Comment by alanmarkoTrilitech** [comment]
I tried running it, but the example doesn't seem to work properly:

`Using mock self-address tz4RepLRepLRepLRepLRepLRepLRepN7Cu8j.
>> Contract.create("export default () => { console.log(`Hello from ${Ledger.selfAddress} 👋`); return new Response()) }");
[📜] Contract created: tz4RVBAHakYgHx7XYK1nacqCGikzWBGGm6Ld
tz4RVBAHakYgHx7XYK1nacqCGikzWBGGm6Ld
>> new Request(`tezos://${_}`);
[object Object]
>> Contract.call(_);
Evaluating: "export default () => { console.log(`Hello from ${Ledger.selfAddress} 👋`); return new Response()) }"
Uncaught EvalError: JsError
>> `

**Comment by sam-finch-tezos** [comment]
> I tried running it, but the example doesn't seem to work properly:

There was an extra bracket in the example contract. It should work now

**Review by alanmarkoTrilitech — APPROVED** [review]
LGTM!

File: jstz_cli/src/repl.rs [hunk 1]
```diff
@@ -110,19 +110,32 @@ fn evaluate(input: &str, rt: &mut Runtime, hrt: &mut (impl HostRuntime + 'static
...
-                        .to_std_string_escaped()
-                );
+    jstz_core::future::block_on(async move {
+        match rt_output {
+            Ok(mut res) => {
+                if res.is_promise() {
+                    if let Ok(promise_result) = rt.resolve_value(&res).await {
+                        res = promise_result;
+                    }
+                }
```
Comment (johnyob) [inline jstz_cli/src/repl.rs:120]: `resolve_value` will handle the `is_promise` logic. Can you move this to where we define `rt_output`

**Comment by johnyob** [comment]
LGTM, please clean up your git history before merging (merge commits aren't allowed in `main`)

**Comment by johnyob** [comment]
Changes look good. I'm not entirely happy that we automatically resolve promises (since I don't expect this from most REPLs). But for now it is a good workaround the issues we've had with promises in the REPL.

**Review by johnyob — APPROVED** [review]

**Pull request merged**